modules/emailer/functions/emailer.inc

```diff
@@ -0,0 +1,472 @@
+        if ($data->access !== null) {
+            $sql = "DELETE FROM `EmailListAccess` WHERE `EmailListID` = $id";
+            \DB::run($sql);
+            foreach ($data->access as $rule) {
+                $id = $rule->DepartmentID;
+                $pos = $rule->PositionID;
+                $elist = $rule->EmailListID;
+                $edit = $rule->EditList;
+                $change = $rule->ChangeAccess;
+                $sql = <<<SQL
+                    INSERT INTO `EmailListAccess`(
+                        `DepartmentID`, `PositionID`, `EmailListID`,
+                        `EditList`, `ChangeAccess`
+                    )
+                    VALUES
+                    (
+                        $id, $pos, $elist, $edit, $change
+                    )
+                    ON DUPLICATE KEY UPDATE
+                        `EditList` = $edit,
+                        `ChangeAccess` = $change
+SQL;
+                \DB::run($sql);
+            }
+        }
+
+    }
+
+
+    public function getAccessControl($listID)
+    {
+        global $Departments;
+
+        $output = [];
+        $sql = <<<SQL
+        SELECT
+            (
+                SELECT
+                    `Name`
+                FROM
+                    `Departments`
+                WHERE
+                    `A`.`DepartmentID` = `DepartmentID`
+            ) AS Department,
+            (
+                SELECT
+                    `Name`
+                FROM
+                    `ConComPositions`
+                WHERE
+                    `A`.`PositionID` = `PositionID`
+            ) AS Position,
+            `A`.*
+        FROM
+            `EmailListAccess` as A
+        WHERE
+            `EmailListID` = $listID
+SQL;
+        $result = \DB::run($sql);
+        $value = $result->fetch();
+        if ($value === false) {
+            $sql = "SELECT `COLUMN_NAME` FROM `INFORMATION_SCHEMA`.`COLUMNS` WHERE `TABLE_SCHEMA`='".$GLOBALS['DBNAME']."' AND `TABLE_NAME`='EmailListAccess';";
+            $result = \DB::run($sql);
+            $data = ['Department' => null,
+            'Position' => null];
+            $value = $result->fetch();
+            while ($value !== false) {
+                $data[$value['COLUMN_NAME']] = null;
+                $value = $result->fetch();
+            }
+            $output[] = $data;
+        } else {
+            while ($value !== false) {
+                $output[] = $value;
+                $value = $result->fetch();
+            }
+        }
+        return $output;
+
+    }
+
+
+    public function send($data)
+    {
+        $to = $this->getToById($data->to);
+        $headers = 'From: '.$data->from."\r\n";
+        $headers .= 'Reply-To: '.$data->reply;
+        $headers .= "MIME-Version: 1.0\r\n";
+        $headers .= "Content-Type: text/html; charset=ISO-8859-1\r\n";
+        if (count($to) > 0) {
+            foreach ($to as $target) {
+                mail($target, $data->subject, $data->body, $headers);
```

Comment:
So: since PHP can be configured so that `mail` uses SMTP, we should not have to change **anything** if we switch to using Mailgun or something like it with their SMTP-based interface--just make sure our PHP environment is configured to use it.